consensus/istanbul/backend/api.go

```diff
@@ -327,39 +327,26 @@ func (api *APIExtension) makeRPCBlockOutput(b *types.Block,
 		}
 	}
 
-	committers, _, err := ParseCommitteedSeals(head)
-	if err != nil {
-		parseErr := make(map[string]interface{})
-		parseErr["ERROR"] = err
-		return parseErr
-	}
-
 	r["committee"] = cInfo.Committee
-	r["committers"] = committers
+	r["committers"] = cInfo.Committers
+	r["sigHash"] = cInfo.SigHash
 	r["proposer"] = cInfo.Proposer
 	r["round"] = cInfo.Round
 	r["originProposer"] = cInfo.OriginProposer
 	r["transactions"] = rpcTransactions
 	return r
 }
 
-func ParseCommitteedSeals(header *types.Header) ([]common.Address, [][]byte, error) {
-	if header == nil {
-		return nil, nil, errors.New("Empty header")
-	}
-	istanbulExtra, err := types.ExtractIstanbulExtra(header)
-	if err != nil {
-		return nil, nil, err
-	}
-	committers := make([]common.Address, len(istanbulExtra.CommittedSeal))
-	for idx, cs := range istanbulExtra.CommittedSeal {
-		committer, err := istanbul.GetSignatureAddress(istanbulCore.PrepareCommittedSeal(header.Hash()), cs)
+func RecoverCommittedSeals(extra *types.IstanbulExtra, headerHash common.Hash) ([]common.Address, error) {
+	committers := make([]common.Address, len(extra.CommittedSeal))
+	for idx, cs := range extra.CommittedSeal {
```

Comment:
`nil` check needed

Reply:
I thought it may be good to have a nil-check inside it. if you find callsite checking adequate, go with that.